api/core/EntityMaker.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+abstract class EntityMaker
+{
+    /**
+     * @param array<string,mixed> $data
+     */
+    public static function fill_with_properties(array $data, CrudEntity $entity): void
+    {
+        $entity_name = $entity->get_table_name();
+        $data_properties = array_keys($data);
+        $entity_properties = get_class_vars($entity_name);
+        foreach($data_properties as $d_property) {
+            if (!array_key_exists($d_property, $entity_properties)) {
+                throw new Exception("Attribute \"$d_property\" not found in \"".$entity_name."\"", 500);
+            } else {
+                $entity_property_type = EntityMaker::get_entity_property_type($d_property, $entity_name);
+                if (gettype($data[$d_property]) != $entity_property_type) {
+                    if ($cast = EntityMaker::cast($data[$d_property], $entity_property_type)) {
+                        $data[$d_property] = $cast;
+                    } else {
+                        throw new Exception(
+                            "Property \"$d_property\" must be of type \"$entity_property_type\". "
+                            .gettype($data[$d_property])." given.",
+                            500
+                        );
+                    }
+                }
+                $entity->$d_property = $data[$d_property];
```

Comment:
Entity come from data then is instenciated inside the corresponding entity.
Might need to put the instanciation part somwhere else